src/object/obj_enum.c

```diff
@@ -384,8 +384,21 @@ fill_rec(daos_handle_t ih, vos_iter_entry_t *key_ent, struct dss_enum_arg *arg,
 	if (inline_data && data_size > 0) {
 		d_iov_t iov_out;
 
-		/* inline packing for the small recx located on SCM */
-		D_ASSERT(key_ent->ie_biov.bi_addr.ba_type == DAOS_MEDIA_SCM);
+		/* For SV case, inline data must be located on SCM.
+		 * For EV case, the inline data may be only part of
+		 * the original extent. The other part(s) of the EV
+		 * may be invisible to current enumeration. Then it
+		 * may be located on SCM or NVMe.
+		 */
```

Comment:
This assert was true because old rebuild did physical record enumeration, when rebuild is changed to logical view rebuild, it's not true anymore.

This comment about SV, EV could be removed if the patch needs be refreshed.